### PR TITLE
Prepare for Renovate with reusable workflows

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
Same feedback as https://github.com/mezzio/mezzio-swoole/pull/89#pullrequestreview-888416688

In addition to that, on the laminas side, we need to create the `3.11.x` branch first.

File: composer.json [hunk 1]
```diff
@@ -26,6 +26,9 @@
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true,
             "composer/package-versions-deprecated": true
+        },
+        "platform": {
```
Comment (Ocramius) [inline composer.json:30]: Doesn't the lock file need hash adjusting though?

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Review by Ocramius — APPROVED** [review]
Thanks @ghostwriter - updating base branch + rebase now up to us :)

**Review by weierophinney — APPROVED** [review]

**Comment by Ocramius** [comment]
@weierophinney your opinion? `3.10.x` or `4.0.x`? I don't know if `4.0.x` is active/to be used

**Comment by weierophinney** [comment]
3.x, please. Not sure why we have the v4 branch in place...

**Review by geerteltink — APPROVED** [review]

**Comment by Ocramius** [comment]
Tried changing to `4.0.x`, and commit range is clean there, but @weierophinney asked for `3.x`, so we need to cherry-pick these changes out :|

**Review by Ocramius — CHANGES_REQUESTED** [review]
Let's rebase/cherry-pick on top of `3.10.x` - I'll create `3.11.x` before merging, when at a PC.

**Comment by boesing** [comment]
I've created 3.11.x for #110 and changed the target branch here as well.
Also changed default branch to 3.11.x (was 4.0.x because of the PSR-17 ongoing work).

**Review by boesing — APPROVED** [review]
Rebased against 3.11.x, LGTM

**Review by Ocramius — APPROVED** [review]
Awesome, thanks @ghostwriter and @boesing!

**Pull request merged**